README.md

```diff
@@ -1,7 +1,124 @@
 
+### **Reasons to Choose Ollama**
+
+- **Data Privacy:** Many corporations have privacy regulations that prohibit sending internal code or data to third-party services.
+- **Generated Material Licensing:** Many models, even those with permissive usage licenses, do not disclose their training data and therefore may produce output that is derived from training material with licensing restrictions.
+- **Cost:** Many of these tools are paid solutions that require investment by the organization. For larger organizations, this would often include paid support and maintenance contracts, which can be extremely costly and slow to negotiate.
+
+### Models and Workflow of Vscode-Granite
+
+vscode-granite uses the Granite Code model, which is optimized for enterprise software development workflows and performs well across a range of coding tasks (e.g., code generation, fixing, and explanation), making it a versatile "all-around" code model.
+
+Granite Code comes in a wide range of sizes to fit your workstation's available resources. Generally, the bigger the model, the better the results.
+
+**Recommendation:** Model Size 20B for chat, 8B for tab code completion.
+
+For more details, refer to [Granite Code Models](https://github.com/ibm-granite/granite-code-models)
+
+ ### Workflow of Vscode-Granite
+
+ <p align="center">
+<img src="/media/workflow.png" alt="workflow" width="400"/>
+ </p>
+ 
+### Installation Prerequisites:
+
+  * OS: MacOS 
+  * DISK SPACE :Minimum 30 GB 
+  * latest [Visual Studio Code](https://code.visualstudio.com/)
+  * [Node.js](https://nodejs.org/) v20.0.0 or higher
+  * optional: [gh](https://cli.github.com/) to manage pull requests.
+  * Install [continu.dev](https://marketplace.visualstudio.com/items?itemName=Continue.continue) extention on Visual studio code
+
+### Steps run the extension
+  * clone this repository
+
+  <body>
+    <h4>How to install vs-granite on macOS/Linux</h4>
+    <p>Follow these steps to install <code>vs-granite</code> on your Mac/Linux machine :</p>
+    <ol>
+        <li>Clone the repository:</li>
+        <pre><code>git clone https://github.com/redhat-developer/vscode-granite.git </code></pre>
+        <li>Navigate to the project directory:</li>
+        <pre><code>cd vscode-granite</code></pre>
+        <li>Install the NPM dependencies:</li>
+        <pre><code>npm run install:ci</code></pre>
+    </ol>
+</body>
+
+This will install the dependencies for the extension and its [webview(s)](https://code.visualstudio.com/api/extension-guides/webview)
+
+#### Use Vscode-granite UI to install extention and models
+
+This project offers an intuitive and streamlined UI designed to simplify the installation and management of extensions and Granite models. This user-friendly interface enables developers to quickly set up and configure their environment without needing to dive into complex configurations or terminal commands
+
```

Comment:
The first time the vscode-granite extension starts, the Granite Code setup wizard will open. You can open this wizard any time, from the command palette, execute the "Granite: Setup Granite Code as code assistant" command.